GraphQL/src/schema.py

```diff
@@ -0,0 +1,116 @@
+import strawberry
+from typing import Optional
+import requests
+
+
+@strawberry.type
+class User:
+    username: str
+    email: str
+
+
+@strawberry.type
+class Subscription:
+    user_id: int
+    period: str
+    subscription_type: str
+
+
+@strawberry.type
+class AddUserResponse:
+    status: str
+    message: str
+    user: Optional[User]
+
+
+@strawberry.type
+class UpdateUserResponse:
+    status: str
+    message: str
+
+
+@strawberry.type
+class AddSubscriptionResponse:
+    status: str
+    message: str
+    subscription: Optional[Subscription]
+
+
+@strawberry.type
+class DeleteUserResponse:
+    status: str
+    message: str
+
+
+@strawberry.type
+class Query:
+    @strawberry.field
+    async def get_user_details(self, user_id: int) -> Optional[User]:
```

Comment:
can we use the
```python
user_details: User | None = strawberry.field(resolver=<function ref to which does the requesting>)
```
This way we can abstract the "endpoint" logic to an place specificly dedicated to it and not having it in the graphql Query class inline

Reply:
You should have a second file which will perform the request and return a response which can represent the User object. Strawberry will automatically make this possible. In this variation the parameters written in the function which is referenced as a resolver will be the params that will be expected in the query sent using Postman.

Reply:
The same applies to class Mutation and all its methods